### DISCUSSION TITLE
Aggregations by a combination of fields

### DISCUSSION THREAD

**Discussion by krivulcik** [ravendb/ravendb#12996] [Q&A] [answered]
Is it possible to do the aggregations by a combination of fields (or ranges)?

Inspired by the example here (https://ravendb.net/docs/article-page/5.2/csharp/client-api/session/querying/how-to-perform-a-faceted-search#example-iv):

```
from index 'Camera/Costs'
select
facet(Manufacturer,
      min(Cost),
      max(Megapixels),
      max(Zoom)),
facet(Cost < 400,
      Cost >= 400,
      avg(Cost),
      max(Cost),
      min(Cost))
```

I want to get a matrix of Manufacturer x [lowCost, highCost] combinations and show a table of the number of models.

I imagine a result like the following:
```
Models by:
Manufacturer \ Price range    Cost < 400    Cost >= 400
Canon                                  1              2
Nikon                                  3              4
Sony                                   5              6
```

Is it possible to do in a single query?

Something like

```
from index 'Camera/Costs'
select
facet([COMBINATION OF MANUFACTURER AND COST RANGE])
```

Or do we need to issue separate queries for all of the values in one of the categorizations? In this case, it would be two queries:

```
from index 'Camera/Costs'
where Cost < 400
select
facet(Manufacturer)

from index 'Camera/Costs'
where Cost >= 400
select
facet(Manufacturer)
```

But having a way to retrieve this in one query would be great. In our real-life scenario, the number of facet values is higher. They are also real values in the fields, not ranges, but that's likely not important.

**Comment by ayende** [comment, marked as answer]
You cannot do a facets on the combination of fields, I'm afraid. 
What you can do it use lazy queries to send a *bunch* of queries in a single remote call to the server.

**Reply by krivulcik** [reply]
Yes, that's what I was planning to do, thanks.
